### PR TITLE
Improved_documentation

### LINKED ISSUE #1: Improve Documentation
## Description:
The current documentation for the snapshell project needs enhancements to provide better guidance and clarity for users. The following areas should be addressed:

- Installation Guide: Provide detailed steps for installing the project, including dependencies and environment setup.
- Usage Examples: Add comprehensive examples demonstrating how to use the tool effectively.
- Configuration Instructions: Explain how to configure the tool, including setting environment variables like GROQ_API_KEY.
- API Documentation: Document the API endpoints and their usage if applicable.
- Contribution Guidelines: Include guidelines for contributing to the project, such as code style, testing, and submission processes.
- Troubleshooting: Add a section for common issues and their resolutions.

## Additional Notes:

- Ensure all documentation is up-to-date with the latest version of the tool.
- Use clear and concise language.
- Include screenshots or diagrams where helpful.

### PR DESCRIPTION
**Pull request by nerdylua**
This PR improves the current documentation of the repository. 
Fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,13 +11,58 @@ SnapShell is a powerful command-line utility designed to enhance your Linux term
...
+pip install -r requirements.txt
+python setup.py install
+```
+## Dependencies
 
-## Usage
+SnapShell requires the following libraries:
+
+- `argparse`: For parsing command-line arguments.
+- `colorama`: For colorized terminal output.
```
Comment (KTS-o7) [inline README.md]: we dont use colorama. needs to be changed.

File: README.md [hunk 2]
```diff
@@ -11,13 +11,58 @@ SnapShell is a powerful command-line utility designed to enhance your Linux term
...
+- `sqlite3`: For local database operations.
+- `os`: To interact with your operating system.
+
+## Configuration Instructions
+To use SnapShell, you need to set up the `GROQ_API_KEY` environment variable. This is required for connecting to the underlying Language Model API.
+### Setting the `GROQ_API_KEY`
+1. Obtain your API key from the GROQ service.
+1. Set it as an environment variable:
+```sh
```
Comment (KTS-o7) [inline README.md]: we need to export as HELPER_GROQ_API_KEY. Read through the code base for details.

File: README.md [hunk 3]
```diff
@@ -11,13 +11,58 @@ SnapShell is a powerful command-line utility designed to enhance your Linux term
...
+export GROQ_API_KEY="your_api_key_here"
```
Comment (KTS-o7) [inline README.md]: same as above.

File: README.md [hunk 4]
```diff
@@ -40,53 +85,59 @@ snapshell --update-db
...
+## Troubleshooting
+### Common Issues
+1. **Installation Errors**: Ensure that you have the correct version of Python and pip installed. If dependencies fail to install, try updating pip:
+```sh
+pip install --upgrade pip
 ```
-
+2. **Missing GROQ_API_KEY**: If SnapShell is not functioning properly, make sure the `GROQ_API_KEY` is correctly set as an environment variable. You can check this by running:
+```sh
+echo $GROQ_API_KEY
```
Comment (KTS-o7) [inline README.md]: modify the env variable name.

**Review by KTS-o7 — CHANGES_REQUESTED** [review]
Have added the changes to be made. Please make the necessary changes and submit for review again

**Review by nerdylua — COMMENTED** [review]
Issues fixed

File: README.md [hunk 5]
```diff
@@ -40,53 +84,59 @@ snapshell --update-db
 snapshell --view-history
 snapshell --clear-history
 ```
+## Advanced Usage
+You can chain arguments or integrate SnapShell with other terminal utilities for more advanced operations. Here's an example:
+```sh
+snapshell --update-db | grep "package-name"
```
Comment (KTS-o7) [inline README.md]: We dont have such functionality but can be implemented next

**Review by KTS-o7 — COMMENTED** [review]
We have changed the way to access API. We dont use env variable any more. We are reading it from a config file. Will be adding that

**Review by KTS-o7 — APPROVED** [review]

**Comment by KTS-o7** [comment]
I have made a few more changes to your PR @nerdylelouch . Please read the code more thoroughly and use proper ways to handle file paths and debugging steps. 
For now I have made those changes and commited.
Congratulations on your first merge !

**Pull request merged**